src/utils/text.ts

```diff
@@ -60,3 +60,74 @@ export function getMatches(regex: RegExp, input: string, group = 0) {
 export function getStringSize(value: string) {
     return value.length * 2;
 }
+
+
+export function formatCSS(text: string) {
+
+    function getShift(deep: number) {
+        if (deep === 0) {
+            return '';
+        }
+        if (deep === 1) {
+            return ' '.repeat(4);
+        }
+        if (deep > 1) {
+            return ' '.repeat(4).repeat(deep);
+        }
+    }
+
+    const groups = [
+        '\/\*[\s\S]*?\*\/', // Comment
+        '\[.*\]', // Braces
+        '\(.*\)', // Brackets
+        '".*?"', // single line
+        '\s{2,}', // multiple spaces
+        '\s*\{\s*', // Opening bracket
+        '\s*\}\s*', // end bracket
+        ';\s*' // Normal line
+    ];
+
+    const depth = 0;
+    return text.replace(new RegExp(groups.map(group => `(${group})`).join('|'), 'g'), (match, comment, brace, bracket, single, multi, open, end, normal) => {
+        if (multi) {
+            return ' ';
+        }
+        if (comment || brace || bracket || single) {
+            return match;
+        }
+        if (open) {
+            return getShift(depth) + open.replace(/^\s+/, '') + '\n';
```

Comment:
Don't use replace in places like this, simply put `{` or `}` or `;`

Reply:
Oh this is just trimleft() to trim all already indent.

Reply:
trimLeft/trimStart, but Typescript don't recognize it?